src/foam/util/SafetyUtil.java

```diff
@@ -202,6 +203,17 @@ public static Object deepClone(Object o) {
     }
   }
 
+  /**
+   * @param l1 The list that will be tested if it's a subset
+   * @param l2 The list that l1 will be tested against
+   * @param <T> The type of Object in the lists
+   * @return Whether or not l1 union l2 is equal to l1
+   */
+  public static <T> boolean isSubSet(List<T> l1, List<T> l2) {
+    if ( l1 == null || l2 == null ) return false;
```

Comment:
What if l1 == null && l2 == null?

Reply:
Intentionally left that as false... I figured it would make sense to say if any list is null then return false, because the input is invalid, but if you think treating a null list as an empty list would make more sense I can change it to be so.

Reply:
It would make more sense (to me) to be true, but then we would need to decide what to do for (null, not null) and (not null, null). Given that the semantics don't make much sense, I'm not sure if we want to put this in SafetyUtil, and just add a suitable check where you make this call, since I can't think of a strong case for a standardized semantics.